tools/masspix/masspix.xml

```diff
@@ -0,0 +1,729 @@
+                 label="Specify cluster numbers"
+                 help="Number of clusters to use" />
+
+          <param name="intensity" type="boolean" 
+                 truevalue="TRUE" falsevalue="FALSE" checked="true"
+                 label="Save cluster intensity values"
+                 help="Save cluster intensity for inspection."/>
+      </when>
+
+      <when value="FALSE" />
+
+    </conditional>
+  </inputs>
+
+  <!-- =============================================================== -->
+  <outputs>
+    <data format="tabular" name="image_out"
+          label="Processed imaging data on ${on_string}">
+      <filter> process['process_selector'] == "TRUE" </filter>
+    </data>
+    
+    <data format="rdata" name="rdata_out"
+          label="R running results on ${on_string}">
+          <filter> 
+            ((
+            process['process_selector'] == "TRUE" and
+            process['rdata'] == True
+            ))
+          </filter>
+    </data>
+
+    <data format="pdf" name="pca_out"
+          label="PCA plots on ${on_string}">
+      <filter> pca['pca_selector'] == "TRUE" </filter>
+    </data>
+
+    <data format="tabular" name="loading_out"
+          label="PCA loading file on ${on_string}">
+          <filter> 
+            ((
+            pca['pca_selector'] == "TRUE" and
+            pca['loading'] == True
+            ))
+          </filter>
+    </data>
+
+    <data format="pdf" name="slice_out"
+          label="Slice plots on ${on_string}">
+      <filter> slice_selector = True </filter>
+    </data>
+
+    <data format="pdf" name="clus_out"
+          label="Cluster plots on ${on_string}">
+      <filter> cluster['cluster_selector'] == "TRUE" </filter>
+    </data>
+
+    <data format="tabular" name="intensity_out"
+          label="Cluster intensity file on ${on_string}">
+          <filter> 
+            ((
+            cluster['cluster_selector'] == "TRUE" and
+            cluster['intensity'] == True
+            ))
+          </filter>
+    </data>
+
+  </outputs>
+
+  <!-- =============================================================== -->
+  <!-- 'WriteXLS' gives different size although the contents are the same -->
+  <!-- different result in clustering because of randomness-->
+  <tests>
+    <!-- full process image -->
+    <test expect_num_outputs="7">
+      <expand macro="infile_imzml"/>
+      <param name="process_selector" value="TRUE" /> 
+      <param name="rem_outliers" value="TRUE" /> 
+      <param name="summary" value="TRUE" /> 
+      <param name="rdata" value="TRUE" /> 
+      <param name="pca_selector" value="TRUE" /> 
+      <param name="loading" value="TRUE" /> 
+      <param name="slice_selector" value="TRUE" /> 
+      <param name="cluster_selector" value="TRUE" /> 
+      <param name="intensity" value="TRUE" /> 
+      <output name="image_out" file="res/image.tsv" />
+      <output name="rdata_out" file="res/r_running.rdata" compare="sim_size" />
```

Comment:
can we remove this line, and also the test data it is really large and testing this file, does not mean much.